Orthtree/doc/Orthtree/Concepts/OrthtreeTraits.h

```diff
@@ -18,45 +17,109 @@ class OrthtreeTraits
-  typedef unspecified_type FT; ///< The number type of the %Cartesian coordinates of types `Point_d`
-  typedef unspecified_type Point_d; ///< Point type.
-  typedef unspecified_type Sphere_d; ///< The sphere type for neighbor queries.
+  using Dimension = unspecified_type; ///< Dimension type (see `CGAL::Dimension_tag`).
+  using FT = unspecified_type; ///< The number type of the %Cartesian coordinates of types `Point_d`
+  using Point_d = unspecified_type; ///< Point type.
+  using Bbox_d = unspecified_type; ///< Bounding box type. Must be constructible from a pair of `Point_d` types.
 
   /*!
     A random access iterator type to enumerate the
     %Cartesian coordinates of a point.
   */
-  typedef unspecified_type Cartesian_const_iterator_d;
-  typedef std::array<FT, Dimension::value> Array; ///< Array used for easy point constructions.
+  using Cartesian_const_iterator_d = unspecified_type;
 
-  typedef unspecified_type Adjacency; ///< Specify the adjacency directions
+  /*!
+   * \brief The data type contained by each node.
+   */
+  using Node_data = unspecified_type;
 
   /*!
-    Functor with an operator to construct a `Point_d` from an `Array` object.
-  */
-  typedef unspecified_type Construct_point_d_from_array;
+   * \brief Number-type which can take on values indicating adjacency directions.
+   *
+   * Must be able to take on values ranging from 0 to the number of faces of the (hyper)rectangle type, equivalent to 2 * D.
+   */
+  using Adjacency = unspecified_type; ///< Specify the adjacency directions
 
   /*!
-    Functor with an operator to construct a `Bbox_d` from two `Array` objects (coordinates of minimum and maximum points).
-  */
-  typedef unspecified_type Construct_bbox_d;
+   * \brief Functor with an operator to create the bounding box of the root node.
+   *
+   * The bounding box must enclose all elements contained by the tree.
+   * It may be tight-fitting. The orthtree constructor produces a bounding box surrounding this region.
+   * For traits which assign no data to each node, this can be defined to return a fixed region.
+   */
+  using Construct_root_node_bbox = unspecified_type;
+
+  /*!
+   * \brief Functor which initializes the contained elements for the root node.
+   *
+   * Each node of a tree has an associated `Node_data` value.
+   * For most nodes, this is set by `Distribute_node_contents`, but that is not possible for the root node.
+   * Instead, this functor initializes the `Node_data` of the root node.
+   * It should take no arguments, and return an instance of `Node_data`.
+   *
+   * Typically, the `Node_data` of the root node contains all the elements in the tree.
+   * For a tree in which each node contains an `std::span()` this function would return the span containing all items.
+   *
+   */
+  using Construct_root_node_contents = unspecified_type;
+
+  /*!
+   * \brief Functor to locate in which halfspace a number of type `FT` is located with respect to another number of type `FT`.
```

Comment:
Looking at the implementation in `Orthtree_traits_base_for_dimension`  this looks just like a `Less`  functor (which we do not have in `RealEmbeddableTraits`, but where we have a `Compare`).

Reply:
removed